012_IP_L139/012_IP_L139_safe.py

```
@@ -0,0 +1,333 @@
+    dec=df['description']
+    return render_template('imform.html',page=int(page),username=cookie,tag=tag,dec=dec,img=img,gs=len(tag)+5//6,id=id_,l=len(tag))
+@app.route('/imform/<i>')
+def bird_inet(i):
+    with open(r'/Web/imform.json','r',encoding='utf-8') as f:
+        json=f.read()
+    df = pd.read_json(json, orient='records')
+    tag=df['tag']
+    dec=df['description']
+    img=df['imageUrl']
+    i=int(i)
+    return render_template(f'imform_base.html',name=tag[i],dec=dec[i],img=img[i])
```

Ruff (0.14.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)